### DISCUSSION TITLE
Adding Voice Interface Types (Feature Request #17045)

### DISCUSSION THREAD

**Discussion by mhansen453** [netbox-community/netbox#17074] [Ideas]
Earlier this week I started a feature request (#17045) to include some additional Voice related interface types. There however started to be some discussion about this and I was encouraged to move the discussion over here while the topic gets flushed out.

I am a voice engineer and we have equipment that not only communicates on the data network but also over legacy voice networks. For example I have a Cisco ISR 4331 router that has multiple Ethernet connection but also a NIM expansion module that provides several FXO analog telephone ports. For this reason we have a need to document these legacy voice interfaces in NetBox. 

The first 3 interface types that have come to mind are
- FXO:  Foreign Exchange Office       (used extensively still)
- FXS:  Foreign Exchange Subscriber    (used extensively still)
- E/M:  Earth and Magneto or Ear and Mouth (a much older type (we don't personally use it)

I thought that a new Interface type category called "Voice" could be created with these listed below it. 

I also have a need to document PRI but I figured that out. Since a PRI is just a specialized T1 I used a T1 Interface Type and then mark the circuit I connect to it as type PRI.  That said I might argue that T1, E1, T3, and E3 should also be under Voice but I know they are used for both Voice and data so perhaps designating them in the circuit type is the correct thing to do.

If there are any other Voice Interface Types that think we should ask to be included please add them to the list, or let me know your  thoughts on what I have posted above.

Thanks
Mike

**Comment by kgeorge-PTD** [comment]
I am also trying to add Adtran voice products.  Adding to the drop down under Device Types / Interfaces / "types" it would be helpful to have FXO, FXS, PRI as they are all configureable interfaces on the Adtran's I am entering.
